java/core/src/java/org/apache/orc/impl/BufferChunk.java

```diff
@@ -89,7 +89,10 @@ public boolean equals(Object other) {
       return false;
     }
     BufferChunk ob = (BufferChunk) other;
-    return chunk.equals(ob.chunk);
+    return (chunk != null && ob.chunk != null && chunk.equals(ob.chunk))
+           || (chunk == null && ob.chunk == null
+               && offset == ob.offset
+               && getLength() == ob.getLength());
```

Comment:
This looks like breaking the contact of `equals`.

Reply:
It looks like the offset/length comparison is missing when chunck != null, how about this
```java
return Objects.equals(chunk, ob.chunk) && offset == ob.offset && getLength() == ob.getLength();
```

Reply:
Thank you for reverting, @pavibhai .

Reply:
@dongjoon-hyun I have reverted this and included this comparison logic in the unit test.

IMO it is more thorough to check for offset and length in the equals. We can take that up independent of this patch.

@omalley or @pgaref any thoughts on this?